unblob/handlers/archive/qnap/qnap_nas.py

```diff
@@ -0,0 +1,203 @@
+import io
+from pathlib import Path
+from typing import Optional
+
+import attr
+from dissect.cstruct import Instance
+from pyperscan import Flag, Pattern, Scan, StreamDatabase
+from structlog import get_logger
+
+from unblob.file_utils import File, iterate_file, stream_scan
+from unblob.models import (
+    Endian,
+    Extractor,
+    HexString,
+    StructHandler,
+    StructParser,
+    ValidChunk,
+)
+
+logger = get_logger()
+
+FOOTER_LEN = 74
+SECRET = "QNAPNASVERSION"
+
+C_DEFINITIONS = """
+    typedef struct qnap_header {
+        char    magic[6];
+        uint32  encrypted_len;
+        char    device_id[16];
+        char    file_version[16];
+        char    firmware_date[16];
+        char    revision[16];
+    } qnap_header_t;
+"""
+FOOTER_PATTERN = [
+    HexString("69 63 70 6e 61 73"),
+]
+
+
+@attr.define
+class QTSSearchContext:
+    start_offset: int
+    file: File
+    end_block_offset: int
+
+
+def is_valid_header(header: Instance) -> bool:
+    if header.encrypted_len < len(header):
+        return False
```

Comment:
IIRC the encrypted_len was fixed in all the samples we looked at (`0x10000` if I'm right).

```suggestion
    if header.encrypted_len != FIXED_ENCRYPTED_LEN:
        return False
```

Reply:
According to the PC1 [tool](https://gist.github.com/galaxy4public/0420c7c9a8e3ff860c8d5dce430b2669), it's 0x100000

Reply:
As agreed, just remove lines 48-49.